pkg/apiserver/resthandler.go

```diff
@@ -150,30 +147,15 @@ func getRequestOptions(req *restful.Request, scope RequestScope, kind string, su
 		query = newQuery
 	}
 
-	// TODO Options a mess.  Basically the intent is:
-	// 1. try to decode using the expected external GroupVersion
-	// 2. if that fails, fall back to the old external serialization being used before, which was
-	//    "v1" and decode into the unversioned/legacykube group
-	gvString := scope.APIVersion
-	internalGVString := scope.InternalVersion.String()
-
-	versioned, err := scope.Creater.New(gvString, kind)
+	versioned, err := scope.Creater.New(externalKind.GroupVersion().String(), externalKind.Kind)
 	if err != nil {
-		gvString = "v1"
-		internalGVString = ""
-
-		var secondErr error
-		versioned, secondErr = scope.Creater.New(gvString, kind)
-		// if we have an error, return the original failure
-		if secondErr != nil {
-			return nil, err
-		}
+		return nil, err
 	}
 
 	if err := scope.Codec.DecodeParametersInto(query, versioned); err != nil {
 		return nil, errors.NewBadRequest(err.Error())
 	}
-	out, err := scope.Convertor.ConvertToVersion(versioned, internalGVString)
+	out, err := scope.Convertor.ConvertToVersion(versioned, internalKind.GroupVersion().String())
```

Comment:
> Why do we need the internalKind, why can't we do it the old way, using scope.InternalVersion?

`scope.InternalVersion` is the internal version of the object you're returning.  This method needs to have the internal version of your `getOptions`.  The two are always different except for the legacy kube group.